src/audio/module_adapter/module/generic.c

```diff
@@ -230,7 +253,8 @@ EXPORT_SYMBOL(mod_balloc_align);
  *
  * The allocated memory is automatically freed when the module is unloaded.
  */
-void *mod_alloc_ext(struct processing_module *mod, uint32_t flags, size_t size, size_t alignment)
+void *z_impl_mod_alloc_ext(struct processing_module *mod, uint32_t flags, size_t size,
```

Comment:
I'm wondering if a syscall is really needed here. In theory, all the required structures (`struct processing_module`, `struct module_resources`) are placed in memory accessible to the user. So it should be able to perform allocations on their own heap. The containers are also allocated on the module's heap, so bookkeeping shouldn't be an issue. I assume the problem might be the spinlock operation. Wasn’t there a plan to add a syscall for spinlock operations? On the other hand, is it safe for the kernel to operate on structures inside `struct k_heap` like `_wait_q_t` and `struct k_spinlock` which can be manipulated by the user?

Reply:
Otoh, this is not on the hot path so maybe the syscall is a safer bet to start with.